### PR TITLE
Parking size feature

### PR DESCRIPTION
**Pull request by rodriguezbruno**

### CODE REVIEW

File: app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.example.parkingproyect.mvp.model;
+
+import android.util.Log;
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java]: This import is not being used, please remove it.

File: app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.example.parkingproyect.mvp.view.ParkingView;
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java]: This import is not being used, please remove it.

File: app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      private Parking parking;
+      private ParkingView parkingView;
+
+      public ParkingPresenter(Parking parking, ParkingView parkingView){
+             this.parking = parking;
+             this.parkingView = parkingView;
+      }
+
+      public void onSizeCreationButtonPressed(){
+          //Declaro la variable string para dejarlo mas claro
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java]: Please write your comments in English

File: app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    private final ActivityMainBinding binding;
+
+    public ParkingView(Activity activity, ActivityMainBinding binding){
+       super(activity);
+       this.binding = binding;
+    }
+
+    public String getSize(){
+        String size = "";
+        size = binding.quantityParkingSlots.getText().toString();
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java]: Variable declaration and assignment can be done in one single line.

File: app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        return size;
+    }
+
+    public void showSizeMessage(String size) {
+        binding.sizeLabel.setText("Parking lots added: " + size);
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java]: Avoid hardcoded texts, please define a string resource for easier maintenance.

File: app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public void showInvalidError() {
+        Toast.makeText(getContext(),"Error: " + getSize(),Toast.LENGTH_LONG).show();
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java]: Avoid hardcoded texts, please define a string resource for easier maintenance and use a more descriptive message to explain users what is the issue.

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,18 +1,37 @@
...
+android:layout_height="match_parent"
+android:gravity="center"
+android:orientation="vertical">
+
+
+    <TextView
+        android:id="@+id/parking_size_label"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Parking Size"/>
```
Comment (imachadoglobant) [inline app/src/main/res/layout/activity_main.xml]: Avoid hardcoded texts, please define a string resource for easier maintenance.

File: app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -1,18 +1,37 @@
...
+
+
+    <EditText
+        android:id="@+id/quantity_parking_slots"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:hint="Size..."
```
Comment (imachadoglobant) [inline app/src/main/res/layout/activity_main.xml]: Avoid hardcoded texts, please define a string resource for easier maintenance.

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -1,18 +1,37 @@
...
+    <Button
+        android:id="@+id/addsize_button"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        android:text="Add Size" />
```
Comment (imachadoglobant) [inline app/src/main/res/layout/activity_main.xml]: Avoid hardcoded texts, please define a string resource for easier maintenance.

File: app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.ArrayList;
+
+public class Parking {
+
+    private int parkingSize;
+    public ArrayList<ParkingLot> parkingLotList;
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java]: This should be private.

File: app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -1,18 +1,37 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+android:id="@+id/activity_main"
+android:layout_width="match_parent"
+android:layout_height="match_parent"
+android:gravity="center"
+android:orientation="vertical">
+
+
+    <TextView
+        android:id="@+id/parking_size_label"
```
Comment (imachadoglobant) [inline app/src/main/res/layout/activity_main.xml]: Please review that all your views IDs comply with [Mobile Resources Style for Android](https://docs.google.com/document/d/1eImvh7-1tTKn6e2ivGDChqoD76nw4VzNk_ompvz4kJg/edit#heading=h.5r09p6gu0g3o)

**Review by imachadoglobant — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java (hunk 1 above)
Comment (rodriguezbruno) [reply app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java]: Okey thanks Isma!

File: app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java (hunk 6 above)
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java]: getContext() could potentially return null and then your app would crash, please validate context is not null before using it.

**Review by imachadoglobant — APPROVED** [review]
Looks good!

File: app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+    private int parkingSize;
+    private ArrayList<ParkingLot> parkingLotList;
+
+    public Parking(){
+        this.parkingLotList = new ArrayList<>();
+    }
+
+    public int getParkingSize() {
+            return parkingSize;
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java:8]: please check indentation

File: app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    public void setParkingSize(String newParkingSize) throws NumberFormatException{
+            this.parkingSize = Integer.parseInt(newParkingSize);
+    }
+
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/model/Parking.java]: blank space

File: app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+          String size = parkingView.getSize();
+          try {
+              parking.setParkingSize(size);
+              parkingView.showSizeMessage(String.valueOf(size));
+          }
+          catch(NumberFormatException ex){
+              Log.e(ParkingPresenter.class.getSimpleName(), ex.toString());
+              parkingView.showInvalidError();
+          }
+
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java]: blank space

File: app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+      }
+
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java]: blank space

File: app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+import com.example.parkingproyect.mvp.model.Parking;
+import com.example.parkingproyect.mvp.view.ParkingView;
+
+public class ParkingPresenter {
+
+      private Parking parking;
+      private ParkingView parkingView;
+
+      public ParkingPresenter(Parking parking, ParkingView parkingView){
+             this.parking = parking;
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/presenter/ParkingPresenter.java]: you can use the code format option to improve these kind of indentation problems

I would suggest also to run it on every class

File: app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    private final ActivityMainBinding binding;
+
+    public ParkingView(Activity activity, ActivityMainBinding binding){
+       super(activity);
+       this.binding = binding;
+    }
+
+    public String getSize(){
+        String size = binding.inputParkingQuantitySlots.getText().toString();
+        return size;
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java]: in this case you can directly have something like this:

```
public String getSize(){
        return binding.inputParkingQuantitySlots.getText().toString();
}
```

the main reason is that the variable is not needed anymore, if you return directly in this case, you avoid the garbage collector to recycle the String **size**

File: app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public void showSizeMessage(String size){
+        binding.textParkingMessage.setText(getContext().getString(R.string.text_parking_add_size_message, size));
```
Comment (carlos-navas-globant) [inline app/src/main/java/com/example/parkingproyect/mvp/view/ParkingView.java]: can you check if `getContext()` is actually needed here? I believe you can use `getString()` directly

File: app/src/main/res/layout/activity_main.xml [hunk 19]
```diff
@@ -1,18 +1,37 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".MainActivity">
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (carlos-navas-globant) [inline app/src/main/res/layout/activity_main.xml:2]: same here, formatting

File: app/src/test/java/com/example/parkingproyect/ParkingPresenterTest.java [hunk 20]
```diff
@@ -0,0 +1,21 @@
+package com.example.parkingproyect;
```
Comment (carlos-navas-globant) [inline app/src/test/java/com/example/parkingproyect/ParkingPresenterTest.java]: Empty class?

**Review by carlos-navas-globant — APPROVED** [review]

**Pull request merged**